doctr/models/recognition/vitstr/tensorflow.py

```diff
@@ -202,10 +200,9 @@ def _vitstr(
 
     kwargs["vocab"] = _cfg["vocab"]
 
-    # Feature extractor
-    # NOTE: switch to IntermediateLayerGetter if pretrained vit models are available
+    # feature extractor
     feat_extractor = backbone_fn(
-        pretrained=pretrained_backbone,
+        pretrained=False,  # TODO: pretrained_backbone, solve weights shape mismatch
```

Comment:
@frgfm It is not possible to load the pretrained ViT as backbone for ViTSTR:

File "/home/felix/Desktop/doctr/doctr/models/recognition/vitstr/tensorflow.py", line 122, in call
    features = self.feat_extractor(x, **kwargs)  # (batch_size, patches_seqlen, d_model)
  File "/home/felix/Desktop/doctr/doctr/models/modules/vision_transformer/tensorflow.py", line 91, in call
    patches = self.proj(patches, **kwargs)  # (batch_size, num_patches, d_model)
ValueError: Exception encountered when calling layer "patch_embedding" (type PatchEmbedding).

```
In this `tf.Variable` creation, the initial value's shape ((48, 384)) is not compatible with the explicitly supplied `shape` argument ((192, 384)).

Call arguments received:
  • x=tf.Tensor(shape=(64, 32, 128, 3), dtype=float32)
  • kwargs={'training': 'True'}
```
Any idea how we can solve this ? I have tried to load weights with by_name=True and skip_mismatch=True but it doesn't work we need something we have done in pytorch with ignore_keys any idea if it is possible to do the same in TF ? :thinking:

Reply:
otherwise maybe giving the Dense layer a name could solve the issue for by_name / skip_mismatch :thinking:

Reply:
Mmmh, TF is annoying with this. We'll need to dive into the lower code of the keras API to give you an answer :/

Reply:
solved in a bit different way :) will push it until training finished